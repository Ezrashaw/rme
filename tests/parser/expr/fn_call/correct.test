foo();
5(1);
(1 + 1)(1, 2);
foo()(1)();