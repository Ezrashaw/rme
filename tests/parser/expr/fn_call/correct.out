foo[95m([0m[95m)[0m[90;1m;[0m
[33;3m5[0m[95m([0m[33;3m1[0m[95m)[0m[90;1m;[0m
[95m([0m[33;3m1[0m + [33;3m1[0m[95m)[0m[95m([0m[33;3m1[0m[90;1m, [0m[33;3m2[0m[95m)[0m[90;1m;[0m
foo[95m([0m[95m)[0m[95m([0m[33;3m1[0m[95m)[0m[95m([0m[95m)[0m[90;1m;[0m

---AST Statement (semi: 5..6) #0---
Statement(Expression)@0..5
| FunctionCall@0..5
| | call-expr: Variable("foo")@0..3
| | parens: (3..4, 4..5)
-------

---AST Statement (semi: 11..12) #1---
Statement(Expression)@7..11
| FunctionCall@7..11
| | call-expr: Float(5.0)@7..8
| | parens: (8..9, 10..11)
| | arg #0 (comma): None
| | arg #0: Float(1.0)@9..10
-------

---AST Statement (semi: 26..27) #2---
Statement(Expression)@13..26
| FunctionCall@13..26
| | call-expr: Paren@13..20
| | | parens: (13..14, 19..20)
| | | expr: BinaryOp@14..19
| | | | op: Add@16..17
| | | | lhs: Float(1.0)@14..15
| | | | rhs: Float(1.0)@18..19
| | parens: (20..21, 25..26)
| | arg #0 (comma): Some(22..23)
| | arg #0: Float(1.0)@21..22
| | arg #1 (comma): None
| | arg #1: Float(2.0)@24..25
-------

---AST Statement (semi: 38..39) #3---
Statement(Expression)@28..38
| FunctionCall@28..38
| | call-expr: FunctionCall@28..36
| | | call-expr: FunctionCall@28..33
| | | | call-expr: Variable("foo")@28..31
| | | | parens: (31..32, 32..33)
| | | parens: (33..34, 35..36)
| | | arg #0 (comma): None
| | | arg #0: Float(1.0)@34..35
| | parens: (36..37, 37..38)
-------
